File: fixtures/swift/Sources/Calculator/Calculator.swift
/// A simple calculator that tracks the last result.
public class Calculator {
    private var lastResult: Double = 0.0

    public init() {}

    /// Returns the sum of a and b.
    public func add(_ a: Double, _ b: Double) -> Double {
        lastResult = Operations.add(a, b)
        return lastResult
    }

    /// Returns the difference of a and b.
    public func subtract(_ a: Double, _ b: Double) -> Double {
        lastResult = Operations.subtract(a, b)
        return lastResult
    }

    /// Returns the product of a and b.
    public func multiply(_ a: Double, _ b: Double) -> Double {
        lastResult = Operations.multiply(a, b)
        return lastResult
    }

    /// Returns the quotient of a and b, or throws if b is zero.
    public func divide(_ a: Double, _ b: Double) throws -> Double {
        lastResult = try Operations.divide(a, b)
        return lastResult
    }

    /// The result of the most recent operation.
    public func getLastResult() -> Double {
        return lastResult
    }
}

File: fixtures/go/calculator.go
package calculator

// Calculator performs basic arithmetic operations and tracks the last result.
type Calculator struct {
	lastResult float64
}

// New creates a new Calculator with a zero initial result.
func New() *Calculator {
	return &Calculator{lastResult: 0}
}

// Add returns the sum of a and b.
func (c *Calculator) Add(a, b float64) float64 {
	c.lastResult = Add(a, b)
	return c.lastResult
}

// Subtract returns the difference of a and b.
func (c *Calculator) Subtract(a, b float64) float64 {
	c.lastResult = Subtract(a, b)
	return c.lastResult
}

// Multiply returns the product of a and b.
func (c *Calculator) Multiply(a, b float64) float64 {
	c.lastResult = Multiply(a, b)
	return c.lastResult
}

// Divide returns the quotient of a and b, or an error if b is zero.
func (c *Calculator) Divide(a, b float64) (float64, error) {
	result, err := Divide(a, b)
	if err != nil {
		return 0, err
	}
	c.lastResult = result
	return c.lastResult, nil
}

// LastResult returns the result of the most recent operation.
func (c *Calculator) LastResult() float64 {
	return c.lastResult
}
